src/Common/PathUtilities.cs

```diff
@@ -5,41 +5,14 @@ namespace Microsoft.DotNet;
 
 static class PathUtilities
 {
-    const int S_IRUSR = 256;
-    const int S_IWUSR = 128;
-    const int S_IXUSR = 64;
-    const int S_IRWXU = S_IRUSR | S_IWUSR | S_IXUSR; // 700 (octal) Permissions 
-
-    const int MAX_NUM_DIRECTORY_CREATE_RETRIES = 2;
-
     public static string CreateTempSubdirectory()
     {
-        return CreateTempSubdirectoryRetry(0);
-    }
-
-    [DllImport("libc", SetLastError = true)]
-    private static extern int mkdir(string pathname, int mode);
-    private static string CreateTempSubdirectoryRetry(int attemptNo)
-    {
+#if NETFRAMEWORK
         string path = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
-        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-        {
-            int mkdirStatusCode = mkdir(path, S_IRWXU);
-            if (mkdirStatusCode != 0)
-            {
-                int errno = Marshal.GetLastWin32Error();
-                if (Directory.Exists(path) && attemptNo < MAX_NUM_DIRECTORY_CREATE_RETRIES)
-                {
-                    return CreateTempSubdirectoryRetry(attemptNo + 1);
-                }
-                else
-                    throw new IOException($"Failed to create a temporary subdirectory {path} with mkdir, error code: {errno}");
-            }
-        }
-        else
-        {
-            Directory.CreateDirectory(path);
-        }
+        Directory.CreateDirectory(path);
```

Comment:
Ok, this all looks good to me except this line now. I am checking to see if this is created with 700 permissions or not. If it is then we can merge. I honestly forgot why do we even build for net framework at this point, so I need to check that too.

https://github.com/microsoft/referencesource/blob/master/mscorlib/system/io/directory.cs Looked very windows focused..

Reply:
Ahhh yes that is because this code runs in VS because VS is still perma stuck on framework so we need this 🥱  Ugh ok

Reply:
Sorry this has taken so long, btw. I hope to have a better answer tomorrow on this.